### DISCUSSION TITLE
What do you guys use the `q` command `[@]` register for?

### DISCUSSION THREAD

**Discussion by adriangalilea** [helix-editor/helix#11226] [General]
> q | Play back a recorded macro from the selected register (experimental) | replay_macro

[source](https://docs.helix-editor.com/keymap.html)

I just replaced it as:
```
[keys.normal]
"q" = ":quit"
```

So it's consistent with lazygit, but I'm curious, what am I missing with macros?

Afaik they are great but would be specially great if they didn't disappear between sessions, right?

**Comment by MatrixManAtYrService** [comment]
I've never felt the need to hang onto a macro for more than 5 minutes, maybe that's just because I have the brain of a squirrel.

I use `q` like so:

![2024-09-27 18 08 47](https://github.com/user-attachments/assets/ec9e97fa-b21e-4316-ba61-e68e052684de)

**Reply by adriangalilea** [reply]
I would use multicursor for this but our limited memory is a valid point, thank you for taking the time to record this ;)

**Reply by MatrixManAtYrService** [reply]
There's something pretty satisfying about working with multiple selections, but if they're too widespread I end up overlooking one that turned out to be a special case. I don't do that to myself if I'm focusing on them one at a time, but in rapid succession

**Comment by rahil627** [comment]
there will be a time in which you've got to do something, multiple times, perhaps in multiple files (_which multiple cursors def can't handle_). When it happens, you will be happy to have the keyboard macro feature. 🌝

but i do agree that it doesn't have to be mapped to `q`. could just bind it to `@`, the key of register, or some pop-up menu thing. if really needed, i'd map `c-q` to quit instead.

also that example by the matrixman is a really, really practical one. multiple cursors is great if the content is formatted consistently, perfectly, with no weird lines in-between. more often though, have to jump around files searching for something, and then running the macro.

**Comment by MatrixManAtYrService** [comment]
I have noticed that if you're using macros, and you tab-complete something, the tab completion will not be part of the macro.  Upon replaying it you'll just get whatever you typed before you accepted the completion.

When working with multiple selections, you can tab-complete many at once.  I'm tempted to call this a bug, but maybe it's a necessary evil because tab completions may resolve differently in different places and maybe you don't want that kind of ambiguity when playing a macro?

Do you guys think it's a bug?
